pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -18,6 +19,7 @@ type retentionInterpreter func(desired *unstructured.Unstructured, observed *uns
 
 func getAllDefaultRetentionInterpreter() map[schema.GroupVersionKind]retentionInterpreter {
 	s := make(map[schema.GroupVersionKind]retentionInterpreter)
+	s[appsv1.SchemeGroupVersion.WithKind(util.DeploymentKind)] = retainK8sWorkloadReplicas
```

Comment:
Generally  LGTM.

However, I still think that this is a general solution, not just for `Deployment`.

Reply:
I may not have understood what you meant before, but now I understand that you mean adding a default operation to process all resources and retain the replicas as long as it contain the target label. Am I right?

Reply:
>  you mean adding a default operation to process all resources and retain the replicas as long as it contain the target label. Am I right?

Yes.

Reply:
Agree!

Reply:
done

Reply:
Hi @chaunceyjiang, We just tried it, and setting it to the default path may not work, because one reason is that if we set it to the default path, `HookEnable` is not registered and the existing objectwatcher needs to be adjusted. (If the modification is required, the modification is related to the reconstruction of the `HookEnable` interface.)
Cause 2: For the CRD resource, the replicas of the CRD resource is uncertain. If the CRD resource is set to the default path, some user scalability is lost.

Reply:
So, my suggestion is to keep the original writing first, but I think the `default path thing` is still meaningful, it still needs to be done, and I will add a new task item to the issue to describe the task. If you are interested, you can also assign this task.
How do you think @chaunceyjiang